prov/util/src/util_coll.c

```diff
@@ -150,7 +154,14 @@ int ofi_av_set_insert(struct fid_av_set *set, fi_addr_t addr)
 		if (av_set->fi_addr_array[i] == addr)
 			return -FI_EINVAL;
 	}
+	if (av_set->fi_addr_count >= av_set->max_array_size)  {
+		FI_INFO(av_set->av->prov, FI_LOG_AV,
+            "av_set_insert exceeds av addr array size\n");
+		return -FI_ENOMEM;
+    }
 	av_set->fi_addr_array[av_set->fi_addr_count++] = addr;
+	FI_DBG(av_set->av->prov, FI_LOG_AV,
+            "av_set_insert fi_addr: %" PRIu64 "\n", addr);
```

Comment:
Some notes: Indentation is off in the FI_INFO statements above.  max_array_size is never initialized.  And the check should probably be (fi_addr_count >= max_array_size - 1), depending on what max_array_size is set to.

Reply:
max_array_size is set to allocated array size.  The fi_addr_count is the available # addr in the array, so when array is full, it will have "fi_addr_count == max_array_size".  Do we need to leave an ending space here?